par-script/src/main/java/org/ow2/parscript/PARScriptEngine.java

```diff
@@ -80,34 +81,48 @@
     private final boolean dumpErrorsIfNotForked;
 
     /**
-     * Creates or retrieves a singleton instance of the PARScriptEngine, that wraps an instance of
-     * JRIEngine.
+     * Creates or retrieves a singleton instance of the PARScriptEngine,
+     * that wraps an instance of JRIEngine.
      *
      * @return the singleton instance of the engine
      */
     public static synchronized PARScriptEngine create(PARScriptFactory factory) {
-        if (instance == null) {
-            // Check if the path to rJava is already set
-            String libPath = System.getProperty("java.library.path");
-            if (libPath == null || !libPath.contains("jri")) {
-                try {
-                    RLibPathConfigurator.configureLibraryPath();
-                } catch (Exception e) {
-                    throw new IllegalStateException("Unable to configure the library path for R", e);
-                }
-            }
-            String[] args = { "--vanilla", "--slave" };
+        if (isInForkedTask()) {
+            instance = createScriptEngine(factory);
+        } else if (instance == null) {
+            instance = createScriptEngine(factory);
```

Comment:
The idea is to recreate the script engine at each invocation if _isInForkedTask_ is _true_ but only once if it not the case. 

I agree that the condition could be factorized using a single _if_ with logical operators (probably something like `isInForkedTask() || (instance == null)`. However, using logical operators _with this example_ forces me to draw the _truth table_ to check that it will work as expected.  

I think that code clarity is sometimes more important than really small code duplication. Maybe I am wrong. I let you do the modification if really think it is valuable.